lib/serve-asset-middleware.js

```diff
@@ -0,0 +1,32 @@
+'use strict';
+var serveAsset = require('./utils/serve-asset');
+var setResponseHeaders = require('./utils/response-header').setResponseHeaders;
+
+module.exports = function serveAssetsMiddleware() {
```

Comment:
I don't think `serveAssetsMiddleware` needs to be a function that returns a middleware.  We can just export the middleware directly.

The reason the previous implementation had a function that returned a middleware was to close over the `watcher` argument, but that's not necessary here.

ie we can just `module.exports = function(request, response, next) { // ...`